opencog/reasoning/RuleEngine/rules/relex-to-logic/TOBE.scm

```diff
@@ -0,0 +1,73 @@
+(define TOBE
+	(BindLink
+		(ListLink
+			(TypedVariableLink
+				(VariableNode "$a-parse")
+				(TypeNode "ParseNode")
+			)
```

Comment:
Gosh. Okay. Agaiin, i was just copying your examples; i figure that was
what they were for; to show us what you wanted them to look like!

On Thu, Apr 30, 2015 at 2:24 PM, Linas Vepštas <notifications@github.com>
wrote:

> In opencog/reasoning/RuleEngine/rules/relex-to-logic/TOBE.scm
> <https://github.com/opencog/opencog/pull/1575#discussion_r29406060>:
>
> > @@ -0,0 +1,73 @@
> > +(define TOBE
> > +	(BindLink
> > +		(ListLink
> > +			(TypedVariableLink
> > +				(VariableNode "$a-parse")
> > +				(TypeNode "ParseNode")
> > +			)
>
> All of these patterns are very similar, and are very verbose. They might
> be easier to read if compactified a bit, so e.g. try this:
>
> (define (vardecl a b) (TypedVariableLink (VariableNode a)((TypeNode b)))
>
> Then you could just say (vardecl "$a-parse" "ParseNode") in just one
> line. Or for even more traditional computerese-style, swap a and b...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencog/opencog/pull/1575/files#r29406060>.
>